hooks/expand_show_tree.py

```diff
@@ -0,0 +1,120 @@
+"""
+This hook is called when an engine, app or framework calls 
+
+> self.expand_show_tree(path)
+
+Typically apps, engines and frameworks call this method
+when they want to map the given path on a show tree.
+The default implementation expands this path on the basis of the context of ``bundle_obj``
+And would resolve to either shot/sequence/show depending on which folders exist,
+"$DD_SHOWS_ROOT/$DD_SHOW/$DD_SEQ/$DD_SHOT",
+"$DD_SHOWS_ROOT/$DD_SHOW/$DD_SEQ",
+"$DD_SHOWS_ROOT/$DD_SHOW",
+This would typically not needed to be customized.
+
+In case customization is required, the hook is passed the app/engine/framework
+that issued the original request - this gives access to configuration,
+app methods, environment etc. and should allow for some sophisticated
+introspection inside the hook.
+"""
+
+import os
+
+import sgtk
+from sgtk import dd_xplatform_utils
+
+HookBaseClass = sgtk.get_hook_baseclass()
+
+
+class ExpandShowTree(HookBaseClass):
+    
+    def execute(self, path, bundle_obj, **kwargs):
+        """
+        Handle expansion of the given path on show tree, depending on context issued from an app, framework or engine.
+        
+        :param path: path to create
+        :param bundle_object: object requesting the creation. This is a legacy
+                              parameter and we recommend using self.parent instead.
+        """
+        # this is (:) separated list of paths
+        show_paths = self.expand_tree(bundle_obj, path)
+        paths_list = show_paths.split(":")
+
+        # we need to return the first path since that is the override we have configured
+        # Seq:Shot:Project
+        if paths_list:
+            return paths_list[0]
+
+        # none of the paths exist! the hook was configured without actually populating the show with the hook.
+        return ''
+
+    @staticmethod
+    def expand_tree(bundle_obj, *paths):
+        """Build search paths to form a lookup-tree where each path
+        is appended to a standard location prefix starting from the
+        bottom, walking up to the top.
+
+        >>> self.expand_tree('$DD_OS/solibs', 'lib/xyz').split(":")
+        ['/dd/shows/ABC/RD/cent6/solibs',
+         '/dd/shows/ABC/RD/lib/xyz',
+         '/dd/shows/ABC/cent6/solibs',
+         '/dd/shows/ABC/tools/lib/xyz']
+
+        The generated platform-dependent path is returned with
+        variables expanded to values contained in this instance.
+        The returned path is generated by invoking :meth:`buildPath`.
+
+        """
+        context = bundle_obj.context
+        default_root = bundle_obj.tank.pipeline_configuration.get_primary_data_root()
+        hook_location = "etc/sgtk/hooks"
```

Comment:
I would move this path prefix to __resolve_hooks_path(), and leave this hook generic. That way you can use it for other level path resolutions besides hooks.